lib/service/modules/kinveyDate.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const moment = require('moment');
+
+exports.toKinveyDateString = (date) => {
+  if (typeof date === 'undefined' || date === null) return;
+
+  if (typeof date === 'string' || Object.prototype.toString.call(date) === '[object Date]' ||
+    (moment.isMoment(date) && (typeof date.isValid === 'function' ? date.isValid() : undefined))) {
```

Comment:
I think this part of the `if` can just be `(moment.isMoment(date) && typeof date.isValid === 'function' && date.isValid())`